crates/server/src/server/mod.rs

```diff
@@ -1012,6 +1012,25 @@ pub(crate) async fn handle_request<R: ResponseHandler, T: RequestHandler>(
             addr = src_addr.ip(),
             port = src_addr.port(),
         );
+
+        let Ok(header) = Header::read(&mut decoder) else {
+            return;
```

Comment:
So I guess this is a bad case still, right? Add a comment here maybe like `TODO: should send a response` and some context on why that is non-trivial?

Reply:
I think this is probably fine, since this now only fails on messages that are less than twelve bytes long. I added a comment to this effect.